docs/proposal-rose-suite-run.md

```diff
@@ -24,26 +22,73 @@ The logical step is therefore to move these extra functionalities provided by
 Rose into Cylc - and retire `rose suite-run` and related utilities - so that
 all suites will start up with only Cylc commands in the future.
 
+## Work Plan
+
+- [ ] Agree future form of `cylc-flow.rc` [Cylc-Admin #43](https://github.com/cylc/cylc-admin/issues/43)
+  - [x] Create new issue against cylc-admin
+  - [ ] Create a template `flow.rc`, listing all the options
+    to be available in the combined `flow.rc`. [Link to it](example_files/flow.rc)
+  - [ ] Deprecate settings for job host?
+  - [ ] Add sections for `[job platforms]`
+
+
+- [ ] Implement new `cylc-flow.rc` form. [Cylc-flow #3260](https://github.com/cylc/cylc-flow/issues/3260)
+  - [ ] Check old tests for `global.rc` & `suite.rc` to ensure that functionality
+    is not lost.
+  - [ ] Devise tests for the new flow.rc
+  - [ ] Create new config parser, perhaps called `cylc-flow-rc-parser.py`
+
+
+- [ ] Implement Cluster support functionality. [Cylc-flow #2199](https://github.com/cylc/cylc-flow/issues/2199)
+  - [ ] Modify `task_job_mgr.py` to use the new variables.
+  - [ ] Randomize login host used
+
+
+- [ ] Agree names and functionality of future `cylc flow`(working name)
+  command and command args.[#1030](https://github.com/cylc/cylc-flow/issues/1030)
+  - [ ] Summarize debate on ticket.
+  - [ ] Create a PR against cylc-admin creating a specification for `cylc flow`
+
+
+- [ ] Create a `cylc flow` CLI.
+  - [ ] Include suite validation in `cylc flow` CLI.
+
+
+- [ ] Replace `rose-suite.conf`.
+  - [ ] Create an alternative jinja2 format and tests for this format.
+  - [ ] Create a rose-suite.conf parsing plugin for cylc flow.
```

Comment:
@TomekTrzeciak Suggests that we might prefer to use a python file rather than a jinja2 file as per PR #3169

Reply:
I think we were keen on this approach when last discussed.

Reply:
Yes, discussed in June, as recorded in my meeting notes (not sure who was present at the time): https://github.com/cylc/cylc-admin/pull/37/files#diff-68ceda0f80a6e03c8997fecc75329aafR50

Reply:
note: we will still need to support the `rose-suite.conf` for the foreseeable future (or provide a converter?).

Reply:
Yes, a converter is to be provided. Will add to list.

Reply:
Rose configuration is something that I have been assigned as a main piece of work for now onwards, & I believe this step will come under the remit of my work on that, rather than being part of your ``rose suite-run`` work.

How & when we start to migrate encapsulated suite-level configuration from Rose into Cylc, & how it will fit in, is something that definitely needs a whole discussion on its own, I think. I can't see an Issue for it under ``cylc-flow`` or ``rose``, is anyone aware of one that I may have missed?

Reply:
This here is to ensure that `cylc run` (or whatever) @ Cylc 8 is able to read and process an existing Rose suite (with its `rose-suite.conf` and optional configuration). The work here is to:
* Open up a plugin interface for running 3rd party logic on suite start up.
* Write a plugin that uses the Rose library to interpret and act on suites that still have a `rose-suite.conf` file.

It should not impact on the work on `rose config-edit` - except that the plugin may get to use any new API that will be developed for the `rose config-edit` work.

I guess the wording here should be changed to emphasis the **plugin** bit to avoid confusion?

Reply:
Ah, I see. Thank you @matthewrmshin for clarifying that.

> I guess the wording here should be changed to emphasis the plugin bit to avoid confusion?

Yes, :+1:, a wording change would be helpful, as I clearly interpreted this differently to the intended purpose.

Reply:
For completeness, the PR mentioned is cylc/cylc-flow#3169. In its simplest implementation cylc could just check on suite start for some fixed-name file, like say `suitevars.py` that could be overridden with `--set-file` option. I think this would already provide a lot of the required functionality.

One thing that I would change in that PR is the order in which options are processed and handle `--set FOO=BAR` before (and passed it to) `suitevars.py` or `--set-file=...` - this would allow for conditional logic within the python file based on some command line values. Also, it would be useful to have the suite name (as seen by Cylc for suite registration) passed to this python file as well, so one could configure the suite differently depending on its name (we tend to deploy multiple suites derived from a single source base - it would make it possible to have self-configurable suites with minimal options).